File: MatlabBinaryFiles/countChannels.m
function nC = countChannels(channelMap)
% function nC = countChannels(channelMap)
% count the numebr of set bits in the channel map
nC = 0;
j = 1;
for i = 1:32
    if (bitand(channelMap, j))
        nC = nC + 1;
    end
    j = j * 2;
end

File: HelperFuncs/countChannels.R
# Count the number of set bits in the channel map
# Changed from 32 -> 30 because R hits the maximum integer limit
# There shouldn't ever be that many channels.
countChannels <- function(channelMap) {
    nC <- 0
    j <- 1
    for(i in 1:30) { # Changed, see above
        if(bitwAnd(channelMap, j) != 0) {
            nC <- nC + 1
        }
        j <- j * 2
    }
    return(nC)
}
